eth/fetcher/tx_fetcher.go

```diff
@@ -477,18 +487,26 @@ func (f *TxFetcher) loop() {
 					if waitslots := f.waitslots[ann.origin]; waitslots != nil {
 						waitslots[hash] = ann.metas[i]
 					} else {
-						f.waitslots[ann.origin] = map[common.Hash]*txMetadata{hash: ann.metas[i]}
+						f.waitslots[ann.origin] = map[common.Hash]txMetadata{hash: ann.metas[i]}
 					}
 					continue
 				}
 				// Transaction unknown to the fetcher, insert it into the waiting list
 				f.waitlist[hash] = map[string]struct{}{ann.origin: {}}
-				f.waittime[hash] = f.clock.Now()
+				if ann.metas[i].kind == types.BlobTxType {
+					// blob transactions are never broadcast, so to force them
+					// to be fetched immediately we pretend they arrived
+					// earlier.
+					f.waittime[hash] = f.clock.Now() - mclock.AbsTime(txArriveTimeout)
```

Comment:
Manipulating the time feels wrong. If we want to blob txs to skip the waitlist all together, we should just do that instead of tricking the mechanism to shuffle them out faster than other txs.

Reply:
That was what I implemented in the earlier PR, but @karalabe suggested doing it like this (if I understood him correctly): https://github.com/ethereum/go-ethereum/pull/30118#issuecomment-2209254584
LMK if there's a strong preference either way.

Reply:
@lightclient @karalabe added one more commit which replaces time-travel with explicit skipping of the waitlist if you want to compare.